### PR TITLE
Db 308 override dbconfig

### PR DESCRIPTION
**Pull request by bsweger**
Some changes to support unittests operating in their own copy of the databases:

* Interfaces now reset dbname based on changes that other modules (_e.g._, unit tests) make to the database config settings.
* Add a dropDatabase function to the datatsetSetup script so unit tests can tear-down their temporary databases.
* Update createDatabase function to use the sqlalchemy util database helpful methods instead of manually setting up an engine and checking whether or not a database exists

### CODE REVIEW

File: dataactcore/models/errorInterface.py [hunk 1]
```diff
@@ -6,16 +6,14 @@
 
 class ErrorInterface(BaseInterface):
     """Manages communication with error database."""
-    dbName = CONFIG_DB['error_db_name']
```
Comment (jworcestBAH) [inline dataactcore/models/errorInterface.py]: So moving this into the constructor changes this from a class variable to an instance variable, meaning that doing ErrorInterface.dbName will not work.  I don't remember any specific code using this, but this seems like something we may have done somewhere in the code.

Comment (jworcestBAH) [inline dataactcore/models/errorInterface.py]: So this does break the static getDbName function in each interface, which is being used by env.py

Comment (bsweger) [inline dataactcore/models/errorInterface.py]: Great catch--thanks! Pushed a fix for this and will do the same in the validator repo.

**Comment by jworcestBAH** [comment]
Could we add sqlalchemy_utils to requirements.txt here?

**Comment by bsweger** [comment]
@jworcestBAH pushed fixes for the interfaces and added SQLAlchemy to requirements--my bad, it was already installed in my virtualenv, so just assumed it was already there.

**Comment by jworcestBAH** [comment]
Everything working, merging now.

**Pull request merged**